mvg/http_client.py

```diff
@@ -0,0 +1,74 @@
+import logging
+from requests import Session
+from requests.adapters import HTTPAdapter, Retry
+from mvg.exceptions import raise_for_status
+
+logger = logging.getLogger(__name__)
+
+
+class RequestRetry(Retry):
+    """
+    Retry with logging
+    """
+
+    def __init__(self, *args, **kwargs):
+        if kwargs.get("history"):
+            request = kwargs.get("history")[-1]
+            logger.warning(
+                f"Retring request {request[1]} with status code {request[3]}"
+            )
+
+        super().__init__(*args, **kwargs)
+
+
+class HTTPClient:
+    def __init__(
+        self,
+        endpoint,
+        token,
+        retries=None,
+        timeout=120,
+    ):
+        self.endpoint = endpoint
+        self.token = token
+        self.timeout = timeout
+
+        self.retries = retries
+        if not self.retries:
+            self.retries = RequestRetry(
+                total=3,
+                status_forcelist=[502],
```

Comment:
I would suggest we extend the list to the other common 5xx codes
```suggestion
                status_forcelist=[500, 502, 503, 504],
```

Reply:
Good suggestion. lets do that.